### ISSUE TITLE
Feature: Clarification on requisites/"depend" requisite type

### ISSUE THREAD

**Issue by johnnybubonic** [saltstack/salt#51215]
### Description of Issue/Question
I found the [requisites explanation](https://docs.saltstack.com/en/latest/ref/states/requisites.html) a little hard to follow. Mostly in that based on the explanation, I added a require for an sls and expected it **to be invoked** by the require argument rather than be a require*ment* for the state to be applied.

I suggest clarifying that point. (Take it with a grain of salt, though - ...err, no pun intended - I've been up late so the problem may have been the human parser!)

Additionally, I'd recommend an additional type of requisite, "depend", which would perform the behaviour I expected above (apply a foreign state before applying the current one), as in a "dependency". (If it's possible.) Sort of like prereq (again, assuming I'm understanding it correctly now) except not dependent on a **return** state. This could be useful for states which will always return empty/True(?) - not trigger a prereq, in other words- or for sharing an initiator (such as refreshing an application's cache, etc.). e.g.:

foo.sls:
```YAML
foo_is_active:
  cmd.run:
    - name: /usr/bin/foo
```

bar.sls:
```YAML
run_after_foo:
  cmd.run:
    - name: /usr/bin/read_foo_output
    - depend:
      - sls: foo
```
baz.sls
```YAML
email_bar_results:
  cmd.run:
    - name: /usr/bin/collect_and_send_bar
    - depend:
      - sls: bar
```
In this way, one can apply the "baz" state, and have it chain the dependencies. i.e. baz sees that bar must be run no matter what so it invokes bar, bar sees that etc. for foo. (Obviously one would want to set a limit on the (reverse?) recursion depth.) So you would then have an end result of applying foo, then bar, then baz.

But if I'm misunderstanding the dependency graph currently, then I might have an actual bug on my hands. Let me know and I can open a new issue if so.

### Setup
(N/A)

### Steps to Reproduce Issue
(N/A)

### Versions Report
(N/A)

**Comment by thatch45** [comment]
I think some clarification in the docs would be optimal, but I am hesitant to make a new requisite when we have requisites that already work rather well. Additional closely related requisites seem to only add to user confusion.

**Comment by waynew** [comment]
I can take a look and see if I can come up with some clarification for the docs 🙂

**Comment by johnnybubonic** [comment]
closing; the verbiage is still a little unclear, but the examples help and the requisite system has been fleshed out.
